rust/kernel/task.rs

```diff
@@ -196,6 +196,25 @@ impl Task {
         // running.
         unsafe { bindings::wake_up_process(self.0.get()) };
     }
+
+    /// Tries to give up the CPU and lets another task run.
+    ///
+    /// This maps to kernel's `schedule` function, which is similar to
+    /// [`std::thread::yield_now`].
+    ///
+    /// # Context
+    ///
+    /// This function might sleep, don't call in atomic contexts.
```

Comment:
nit: we're trying to avoid the imperative mode in documentation, so instead of having "do this" or "don't do that", we'd rather have something like "Callers must not be in atomic context because this might sleep".

Reply:
Done locally, thanks!